src/js/internal/stream.ts

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+const ObjectKeys = Object.keys;
+const ObjectDefineProperty = Object.defineProperty;
+
+const customPromisify = require("node:util").promisify.custom;
```

Comment:
maybe we can move this to a separate file so we don't import all of util in here?

Reply:
good point